### PR TITLE
Fix/applics 884 404

### PR DESCRIPTION
**Pull request by HadrianHughesKC**
## Describe your changes

Return 404 and render the 'Not found' page when a project or representation does not exist.

[APPLICS-884](https://pins-ds.atlassian.net.mcas.ms/browse/APPLICS-884?McasTsid=20596&McasCtx=4)

* refactor: remove redundant wrapper functions around API library
  - remove getAppData function and use getProjectData directly
  - remove document.service and use API directly
  - remove registration.service and call API directly
  - remove representation.service and call API directly
  - remove timetable.service
These functions just wrap other functions without adding any other
functionality, so this change removes indirection.
* feat: throw a custom NotFoundError when API returns 404

## Useful information to review or test

Much of this PR is removing the service files and refactoring their consumers to reduce code complexity.

Checked manually that nothing is broken. Update unit tests.

## Type of change 🧩

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have double checked this work does not include any hardcoded secrets or passwords
- [x] I have made corresponding changes to the documentation
- [x] I have provided details on how I have tested my code
- [x] I have referenced the ticket number above
- [x] I have provided a description of the ticket
- [x] I have included unit tests to cover any testable code changes

[APPLICS-884]: https://pins-ds.atlassian.net/browse/APPLICS-884?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by beejjacobs — COMMENTED** [review]
LGTM, will wait for someone more familiar with the app to review/confirm!

File: packages/forms-web-app/src/pages/projects/register/agent/about-project/controller.js [hunk 1]
```diff
@@ -66,7 +63,7 @@ const postRegisterAgentAboutProjectController = async (req, res) => {
 				mode: req.session.mode
 			});
 
-			if (commentsData) await postCommentsData(ipRefNo, commentsData);
+			if (commentsData) await putComments(ipRefNo, commentsData);
```
Comment (Philip-Philippou1) [inline packages/forms-web-app/src/pages/projects/register/agent/about-project/controller.js:66]: Any reason to change this to a PUT?  I am wondering if there could be side affects to this, not sure if PUT returns a reference ID, I think the PUT is deprecated in the API as well.

**Review by Philip-Philippou1 — APPROVED** [review]

**Pull request merged**